app/Services/AWS.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Rogue\Services;
+
+use finfo;
+use Illuminate\Contracts\Filesystem\Filesystem;
+use Illuminate\Filesystem\FilesystemManager;
+use Symfony\Component\HttpFoundation\File\MimeType\ExtensionGuesser;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\Component\HttpKernel\Exception\UnprocessableEntityHttpException;
+
+class AWS
+{
+    /**
+     * The Amazon S3 file system.
+     * @see https://laravel.com/docs/5.2/filesystem
+     * @var Filesystem
+     */
+    protected $filesystem;
+
+    public function __construct(FilesystemManager $filesystem)
+    {
+        $this->filesystem = $filesystem->disk('s3');
+    }
+
+    /**
+     * Store a reportback item (image) in S3.
+     *
+     * @param \Symfony\Component\HttpFoundation\File\UploadedFile|string $file
+     *  File object, or a base-64 encoded data URI
+     * @param string $filename - Filename to write image to
+     *
+     * @return string - URL of stored image
+     */
+    public function storeReportbackItem($file, $filename)
+    {
+        if (is_string($file)) {
+            $data = $this->base64StringToDataString($file);
+        } else {
+            $data = file_get_contents($file->getPathname());
+        }
+
+        $extension = $this->guessExtension($data);
+
+        // Make sure we're only uploading valid image types
+        if (! in_array($extension, ['jpeg', 'png'])) {
+            throw new UnprocessableEntityHttpException('Invalid file type. Upload a JPEG or PNG.');
+        }
+
+        // Add a unique timestamp (e.g. uploads/folder/filename-1456498664.jpeg) to
+        // uploads to prevent AWS cache giving the user an old upload.
+        $path = '/uploads/' . env('S3_BUCKET') . '/' . $filename . '-' . time() . '.' . $extension;
```

Comment:
I think the folder name should be passed in as a parameter (or hard-coded as `reportback-items` if not) here. It doesn't seem super helpful to put the images within a folder named for the bucket they're already being stored in.